tests/foreman/ui/test_hostcollection.py

```diff
@@ -0,0 +1,365 @@
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_host_collection(
+                session, name=name, org=self.organization.name)
+            self.assertIsNotNone(self.hostcollection.search(name))
+            self.hostcollection.update(name, limit='25')
+            self.assertIsNotNone(self.hostcollection.wait_until_element(
+                common_locators['alert.success']))
+
+    @tier1
+    def test_positive_update_limit_to_unlimited(self):
+        """Update Content Host limit from definite number to Unlimited
+
+        @Feature: Host Collection - Positive Update
+
+        @Assert: Host Collection is updated
+        """
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_host_collection(
+                session, name=name, org=self.organization.name, limit='15')
+            self.assertIsNotNone(self.hostcollection.search(name))
+            self.hostcollection.update(name, limit='Unlimited')
+            self.assertIsNotNone(self.hostcollection.wait_until_element(
+                common_locators['alert.success']))
+
+    @tier1
+    def test_negative_update_name(self):
+        """Update existing Host Collection entity name with invalid value
+
+        @Feature: Host Collection - Negative Update
+
+        @Assert: Host Collection is not updated.  Appropriate error shown.
+        """
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_host_collection(
+                session, name=name, org=self.organization.name)
+            self.assertIsNotNone(self.hostcollection.search(name))
+            for new_name in invalid_names_list():
+                with self.subTest(new_name):
+                    self.hostcollection.update(name, new_name=new_name)
+                    self.assertIsNotNone(
+                        self.hostcollection.wait_until_element(
+                            common_locators['alert.error'])
+                    )
+
+    @tier1
+    def test_negative_update_limit(self):
+        """Update Host Collection with invalid Content Host Limit
+
+        @Feature: Host Collection - Negative Update
+
+        @Assert: Host Collection is not updated.  Appropriate error shown.
+        """
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_host_collection(
+                session, name=name, org=self.organization.name)
+            self.assertIsNotNone(self.hostcollection.search(name))
+            for limit in ' ', -1, 'text', '0':
+                with self.subTest(limit):
+                    with self.assertRaises(ValueError) as context:
+                        self.hostcollection.update(name, limit=limit)
+                    self.assertEqual(
+                        context.exception.message,
+                        'Please update content host limit with valid ' +
+                        'integer value'
```

Comment:
Generally we don't assert on `english` texts since we plan to do different language tests in future.